src/Modules/ModulesGraded.jl

```diff
@@ -26,62 +32,394 @@ Graded free module Multivariate Polynomial Ring in x, y over Rational Field grad
   x -> [1]
   y -> [1]
 
+julia> G = grading_group(R)
+GrpAb: Z
+
+julia> graded_free_module(R, [G[1], 2*G[1]])
+Graded free module Multivariate Polynomial Ring in x, y over Rational Field graded by
+  x -> [1]
+  y -> [1]^1([-1]) + Multivariate Polynomial Ring in x, y over Rational Field graded by
+  x -> [1]
+  y -> [1]^1([-2]) of rank 2 over Multivariate Polynomial Ring in x, y over Rational Field graded by
+  x -> [1]
+  y -> [1]
 ```
 """
-function graded_free_module(R::Ring, p::Int, name::VarName=:e, d::Vector{GrpAbFinGenElem}=[grading_group(R)[0] for i in 1:p])
+function graded_free_module(R::Ring, p::Int, W::Vector{GrpAbFinGenElem}=[grading_group(R)[0] for i in 1:p], name::String="e")
+  @assert length(W) == p
   @assert is_graded(R)
   M = FreeMod(R, p, name)
-  M.d = d
+  M.d = W
   return M
 end
 
-function graded_free_module(R::Ring, d::Vector{GrpAbFinGenElem}, name::VarName=:e)
-  p = length(d)
-  return graded_free_module(R, p, name, d)
+function graded_free_module(R::Ring, W::Vector{GrpAbFinGenElem}, name::String="e")
+  p = length(W)
+  return graded_free_module(R, p, W, name)
 end
 
-function graded_free_module(R::Ring, W::Vector{<:IntegerUnion}, name::VarName=:e)
+
+@doc raw"""
+    graded_free_module(R::Ring, W::Vector{<:Vector{<:IntegerUnion}}, name::String="e")
+
+Given a free module `F` over a graded ring with grading group $G = \mathbb Z^m$, and given
+a vector `W` of `ngens(F)` integer vectors of the same size `m`, say, define a $G$-grading on `F` 
+by converting the vectors in `W` to elements of $G$, and assigning these elements as weights to 
+the variables. Return the new module.
```

Comment:
This description may be confusing: 
F does not appear in the arguments and is actually only created and then graded by this function, with ngens(F) taken from the size of the vector W.